rlog/2023-08-04-wakurtosis-waku-scallability.mdx

```diff
@@ -0,0 +1,201 @@
+
+The second plot shows the average peak memory usage with the x-axis also indicating the number of nodes and different traffic loads.
+
+Despite numerous scalability and stability challenges, Wakurtosis has proved effective in simulating a Waku network up to 600 nodes on a single machine, however we encountered issues simulating higher message rates with those network sizes.
+
+#### Without discovery mechanism (baseline)
+
+The analysis reveals interesting trends in bandwidth and memory usage across different network sizes and loads. 
+
+Transmission bandwidth does not consistently increase with more nodes when messages are sent, while reception bandwidth exhibits even more variable behavior depending on scale and load. 
+
+The reduced bandwidth at high message rates stems from simulation infrastructure limits rather than protocol inefficiencies. 
+The no-load measurements provide insights into baseline protocol overhead costs that grow with network size. 
+
+Transmission overhead appears to increase linearly, while reception overhead accelerates sharply beyond 300 nodes. 
+
+![Total average bandwidth usage without discovery mechanism](/static/img/wakurtosis_waku/Baseline_Bandwidth.png) 
+
+<figcaption>
+
+***Total average bandwidth usage without discovery mechanism (baseline).***
+
+</figcaption>
+
+In addition to bandwidth, we also examined average peak memory usage under the different network sizes and messaging loads. 
+
+![Average peak memory usage without discovery mechanism](/static/img/wakurtosis_waku/Baseline_Memory.png) 
+
+<figcaption>
+
+***Average peak memory usage without discovery mechanism (baseline).***
+
+</figcaption>
+
+With no load, memory usage remained consistent around 20-21 MB across all network scales. 
+Under 1 message per second, average memory usage increased slightly to 21-22 MB. 
+The highest memory usage of 23-25 MB was seen with 10 messages per second, especially at 150 nodes. 
+
+This aligns with expectations that higher messaging loads require more memory for message processing and routing. 
+However, the differences are relatively small, with only around a 20% increase from no load to 10 msgs/s. 
+This suggests the protocol has a fairly fixed memory overhead cost, with incremental increases as more messages are handled. 
+Overall, the memory usage appears stable and scales reasonably well, without any dramatic growth as network size expands. 
+This consistency indicates efficient memory management and low overhead from a protocol perspective.
+
+The baseline simulations displayed efficient baseline overhead and scaling for memory, transmission, and reception as message rates grew or network size expanded.
+
+#### With discovery mechanism (Discv5)
+
+With the discovery mechanism, we again see interesting trends in resource usage across different network sizes and traffic loads. 
+
+Transmission bandwidth remainded fairly stable with more nodes.
```

Comment:
What does this mean? Does this tell us something about Waku messaging protocol, our chosen discovery technique and network size?